src/components/HdModal.vue

```diff
@@ -145,151 +97,156 @@ export default {
+    width: #{$sp-l};
+    height: #{$sp-l};
+    margin-bottom: 0;
+    margin-right: $sp-s;
   }
-  &__close {
-    position: absolute;
-    top: $sp-s;
-    right: $sp-s;
-    width: 24px;
-    height: 24px;
-    background-color: transparent;
-    padding: 0;
-    border: 0;
-    cursor: pointer;
-    z-index: 2;
-
-    @media (min-width: $break-tablet){
-      top: $sp-m;
-      right: $sp-m;
-    }
+}
 
-    #{$_root}--external-close-icon & {
-      top: -24px;
-      right: 0;
-      background-color: $primary-color;
-      &::v-deep path {
-        fill: white;
-      }
-    }
+.hd-modal__title {
+  @include font('DS-200');
+  font-weight: bold;
+}
 
-    &--light {
-      &::v-deep path {
-        fill: white;
-      }
-    }
+.hd-modal__body {
+  @include font('DS-100');
+  margin-bottom: $sp-s;
+
+  @media (min-width: $break-tablet) {
+    margin-bottom: $sp-m;
   }
-  &__inner {
-    position: relative;
-    max-width: 100%;
-    margin: auto;
-    z-index: 1;
-    animation: fadeIn .5s forwards, bottomToNormal .5s forwards;
-    background-color: white;
-    border-radius: 3px;
-    box-shadow: 0 5px 40px 0 rgba(0, 0, 0, 0.4);
-
-    #{$_root}--mobile-align-bottom & {
-      margin-bottom: 0;
-
-      @media (min-width: $break-tablet) {
-        margin-bottom: auto;
-      }
+}
+
+.hd-modal__footer {
+  display: flex;
+
+  button {
+    width: 100%;
+
+    &:not(:first-child) {
+      margin-left: $sp-s;
     }
   }
 
-  &__body {
-    background-color: white;
-    padding: $sp-l;
+  @media (min-width: $break-tablet) {
+    justify-content: flex-end;
 
-    #{$_root}--spacing-normal & {
-      padding: $sp-m;
+    button {
+      width: auto;
     }
+  }
+}
+
+.hd-modal__footer--is-wide {
+  flex-direction: column;
 
-    #{$_root}--no-spacing & {
-      padding: 0;
+  button {
+    &:not(:first-child) {
+      margin-left: 0;
+      margin-top: $sp-s;
     }
   }
 
-  &__actions {
-    display: flex;
-    padding: $sp-m ($sp-m - ($sp-xs * 2));
+  @media (min-width: $break-tablet) {
+    flex-direction: row;
 
-    &__action {
-      width: auto;
-      margin-left: $sp-xs;
-      margin-right: $sp-xs;
-      flex-grow: 1;
-      flex-shrink: 0;
-
-      &.btn {
-        box-shadow:
-          0 1px 3px rgba(0, 0, 0, 0.24),
-          0 0 2px rgba(0, 0, 0, 0.12);
-      }
+    button {
+      width: 100%;
 
-      &.btn--ghost {
-        box-shadow:
-          0 1px 3px rgba(0, 0, 0, 0.24),
-          0 0 2px rgba(0, 0, 0, 0.12);
-
-        @media (min-width: $break-tablet) {
-          background-color: transparent;
-          color: getShade($secondary-color, 110);
-        }
+      &:not(:first-child) {
+        margin-left: $sp-s;
+        margin-top: 0;
       }
     }
   }
+}
 
-  &__header {
-    text-align: center;
-    border-bottom: 1px solid getShade($quaternary-color, 60);
+.hd-modal__close-button {
+  position: absolute;
+  right: $sp-m;
+  top: $sp-m;
+  width: #{$sp-m + $sp-s};
+  height: #{$sp-m + $sp-s};
```

Comment:
Nice approach! 
![you-rock](https://user-images.githubusercontent.com/14016028/117282865-7d2d9000-ae65-11eb-98c0-e804d7cf5af0.gif)